rundeckapp/grails-spa/packages/ui-trellis/src/app/components/readme-motd/EditProjectFile.spec.ts

```diff
@@ -0,0 +1,153 @@
+import { mount } from "@vue/test-utils";
+import EditProjectFile from "./EditProjectFile.vue";
+import * as editProjectFileService from "./editProjectFileService";
+
+jest.mock("@/app/components/readme-motd/editProjectFileService", () => ({
+  getFileText: jest.fn().mockResolvedValue({
+    success: true,
+    contents: "sample file content",
+  }),
+
+  saveProjectFile: jest.fn().mockResolvedValue({
+    success: true,
+    message: "File saved successfully.",
+  }),
+}));
+
+jest.mock("@/library/rundeckService", () => ({
+  getRundeckContext: jest.fn().mockReturnValue({
+    rundeckClient: {
+      getFileText: jest.fn().mockResolvedValue({
+        filename: "readme.md",
+        success: true,
+        contents: "Some content",
+        rdBase: "http://localhost:4440",
+      }),
+      saveProjectFile: jest.fn().mockResolvedValue({
+        success: true,
+        message: "File saved successfully.",
+      }),
+    },
+    rdBase: "http://localhost:4440",
+  }),
+  url: jest.fn().mockReturnValue("http://localhost:4440"),
+}));
+let wrapper;
+const mountEditProjectFile = async (props = {}) => {
+  wrapper = mount(EditProjectFile, {
+    props: {
+      filename: "readme.md",
+      project: "default",
+      authAdmin: true,
+      displayConfig: ["none"],
+      ...props,
+    },
+    global: {
+      mocks: {
+        $t: (msg) => msg,
+      },
+    },
+  });
+  await wrapper.vm.$nextTick();
+};
+
+describe("EditProjectFile", () => {
+  beforeEach(async () => {
+    await mountEditProjectFile();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+  it.each([
+    ["readme.md", "edit.readme.label"],
+    ["motd.md", "edit.motd.label"],
+  ])("renders the correct title for %s", async (filename, expectedTitle) => {
+    await mountEditProjectFile({ filename });
+    expect(wrapper.find('[data-test-id="title"]').text()).toContain(
+      expectedTitle,
+    );
+  });
+
+  it("renders file content when getFileText method returns successfully", async () => {
+    expect(wrapper.vm.fileText).toBe("sample file content");
```

Comment:
Could you please change this test to assert instead that the HTML rendered by ace-editor has this string inside of it?